### PR TITLE
Correction for touch layout rotation with layers of differing row counts.

### PR DESCRIPTION
**Pull request by jahorton**
#53 

Includes a very thrown-together keyboard and compatible pages for use in test cases.

### CODE REVIEW

File: web/source/kmwosk.js [hunk 1]
```diff
@@ -2801,6 +2800,11 @@
     
     for(nLayer=0;nLayer<layers.length; nLayer++)
     {
+      // Check the heights of each row, in case different layers have different row counts.
+      nRows=layers[nLayer].childNodes.length;
+      rowHeight=Math.floor(oskHeight/nRows);
```
Comment (darcywong00) [inline web/source/kmwosk.js]: Is it ever possible for layers to not have childNodes and `oskHeight/nRows` becomes a divide by 0?

**Comment by mcdurdin** [comment]
Please include the source of the keyboard you are testing with (i.e. kmn, -layout.js, .kvk if it exists). Perhaps move all the files into a `samples/issue-53` folder given there are so many files.

**Comment by jahorton** [comment]
As requested, I've now provided the full source for the debug keyboard.  It's pretty minimal and specific to #53, so feel free to improve on it.

File: web/source/kmwosk.js (hunk 1 above)
Comment (jahorton) [inline web/source/kmwosk.js]: I doubt that Keyman Developer would let that happen... but a quick check on nRows couldn't hurt.  I did a quick inline `nRows == 0 ? 1 : nRows` in an updated commit.

**Pull request merged**